src/osu_gather.jl

```diff
@@ -0,0 +1,42 @@
+export OSUGather
+
+struct OSUGather <: MPIBenchmark
+    conf::Configuration
+    name::String
+end
+
+function OSUGather(T::Type=Float32;
+                   filename::Union{String,Nothing}="julia_osu_gather.csv",
+                   kwargs...,
+                   )
+    return OSUGather(
+        Configuration(T; filename, max_size=2 ^ 20, kwargs...),
+        "OSU Gather",
+    )
+end
+
+function osu_gather(T::Type, bufsize::Int, iters::Int, comm::MPI.Comm)
+    rank = MPI.Comm_rank(comm)
+    nranks = MPI.Comm_size(comm)
+    send_buffer = ones(T, bufsize)
+    recv_buffer = zeros(T, bufsize * nranks)
+    root = 0
+    timer = 0.0
+
+    MPI.Barrier(comm)
+    for _ in 1:iters
+        tic = MPI.Wtime()
+        if rank == root   
+            MPI.Gather!(MPI.IN_PLACE, UBuffer(recv_buffer, Cint(bufsize) , nranks , MPI.Datatype(T)), comm; root)
```

Comment:
Also this one can be simplified to
```suggestion
            MPI.Gather!(MPI.IN_PLACE, recv_buffer, comm; root)
```

Reply:
When I changed the parameters of the function. It complains that " no method matching Gather!(::MPI.InPlace, ::Vector{Float32}, ::Int64, ::MPI.Comm)".

`ERROR: LoadError: MethodError: no method matching Gather!(::MPI.InPlace, ::Vector{Float32}, ::Int64, ::MPI.Comm)
Closest candidates are:
  Gather!(::Any, !Matched::MPI.UBuffer, ::Integer, ::MPI.Comm) at ~/.julia/packages/MPI/WsAaN/src/collective.jl:249
  Gather!(!Matched::Union{AbstractArray, Ref}, ::AbstractArray, ::Integer, ::MPI.Comm) at ~/.julia/packages/MPI/WsAaN/src/collective.jl:251
  Gather!(::Any, !Matched::Nothing, ::Integer, ::MPI.Comm) at ~/.julia/packages/MPI/WsAaN/src/collective.jl:253
  ...`

Reply:
Right, can use the array if the size can be inferred from `sendbuf`, but in this case it isn't another array.